In [1]:
from xgboost import XGBClassifier
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np
from sklearn.metrics import roc_auc_score, f1_score, recall_score, precision_score, ConfusionMatrixDisplay, RocCurveDisplay
from sklearn.model_selection import StratifiedKFold, train_test_split
from imblearn.over_sampling import SMOTE, ADASYN, SMOTENC
import shap
import optuna


c:\Users\wenyu\AppData\Local\Programs\Python\Python312\Lib\site-packages\tqdm\auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm


In [2]:
df = pd.read_csv("data/train-cat-encoded.csv")
print("Training set size: ", df.shape)
y = df['Will_Buy_EV']
x = df.drop(columns=['Will_Buy_EV'])
x = x.drop(columns=['id', 'Home_Charging_Possible', 'City_Type_Rural', 'Current_Car_Type_Hatchback',
                    'Current_Car_Type_Truck', 'Gender_Male', 'Gender_Female', 'Gender_Other',
                    'Daily_Commute_km', 'Number_of_Cars_Owned', 'Charging_Stations_Near_Home',
                    'Charging_Stations_Near_Work', 'Range_Anxiety_Level'])
binary_features = [
    # 'Home_Charging_Possible', 
    'Subsidy_Available', 
    'City_Type_Urban', 'City_Type_Suburban', 
    # 'City_Type_Rural', 
    'Current_Car_Type_Sedan', 'Current_Car_Type_SUV', 
    # 'Current_Car_Type_Hatchback', 
    # 'Current_Car_Type_Truck', 
    # 'Gender_Male', 'Gender_Female', 'Gender_Other'
    ]
print(binary_features)


Training set size:  (668665, 22)
['Subsidy_Available', 'City_Type_Urban', 'City_Type_Suburban', 'Current_Car_Type_Sedan', 'Current_Car_Type_SUV']


In [3]:
n_splits = 5
skf = StratifiedKFold(n_splits=n_splits, shuffle=True, random_state=42)
subsetNum = 50000
subset_x = x.iloc[:subsetNum]
subset_y = y.iloc[:subsetNum]
smote = SMOTENC(
    random_state=42,
    categorical_features=binary_features)
ada = ADASYN(n_neighbors=5, random_state=42)


In [4]:
hyperparameters = {
    'n_estimators': [50, 100, 200, 300],
    'max_depth': [4, 6, 8, 10],
    'learning_rate': [0.02, 0.05, 0.1, 0.15],
    'subsample': [0.75, 0.8, 0.85],
    'colsample_bytree': [0.75, 0.8, 0.85]
}


In [5]:
X_train, X_test, y_train, y_test = train_test_split(
    subset_x, subset_y, test_size=0.2, random_state=42
)
X_train_resampled, y_train_resampled = smote.fit_resample(X_train, y_train)


In [6]:
print("==========XGBoost Cross Validation Training Loop==========")

def objective(trial):
    n_estimators = trial.suggest_int("n_estimators", 50, 300, step=50)
    max_depth = trial.suggest_categorical("max_depth", [4, 6, 8, 10])
    learning_rate = trial.suggest_categorical("learning_rate", [0.05, 0.1, 0.15])
    subsample = trial.suggest_categorical("subsample", [0.75, 0.8, 0.85])
    colsample_bytree = trial.suggest_categorical("colsample_bytree", [0.75, 0.8, 0.85])
    
    print(f"\n========== n_estimators {n_estimators} max_depth {max_depth}==========")
        
    # 1. Split data (using .iloc if X/y are pandas DataFrames)

    # X_train_resampled, y_train_resampled = ada.fit_resample(X_train, y_train)

    model = XGBClassifier(
    n_estimators=n_estimators,      # number of boosting rounds (trees)
    max_depth=max_depth,           # tree depth
    learning_rate=learning_rate,     # shrinkage per round
    subsample=subsample,         # row sampling per tree
    colsample_bytree=colsample_bytree,  # feature sampling per tree
    eval_metric='logloss'
)
    model.fit(X_train_resampled, y_train_resampled)
    y_pred = model.predict(X_test)
    y_prob = model.predict_proba(X_test)[:, 1]
    
    auc = roc_auc_score(y_test, y_prob)
    f1 = f1_score(y_test, y_pred)
    recall = recall_score(y_test, y_pred)
    precision = precision_score(y_test, y_pred)

    print(f"AUC: {auc:.4f}")
    print(f"F1: {f1:.4f} | Recall: {recall:.4f} | Precision: {precision:.4f}")

    return auc


==========XGBoost Cross Validation Training Loop==========


In [7]:
study = optuna.create_study(direction="maximize")
study.optimize(objective, n_trials=100)


[I 2026-09-11 15:11:35,836] A new study created in memory with name: no-name-069a9b58-6677-4087-a5ae-cb14af8a82b5



========== n_estimators 50 max_depth 4==========


[I 2026-09-11 15:11:36,238] Trial 0 finished with value: 0.9347044949506932 and parameters: {'n_estimators': 50, 'max_depth': 4, 'learning_rate': 0.05, 'subsample': 0.85, 'colsample_bytree': 0.75}. Best is trial 0 with value: 0.9347044949506932.


AUC: 0.9347
F1: 0.6879 | Recall: 0.8238 | Precision: 0.5904

========== n_estimators 200 max_depth 10==========


[I 2026-09-11 15:11:37,763] Trial 1 finished with value: 0.9290901842694801 and parameters: {'n_estimators': 200, 'max_depth': 10, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 0 with value: 0.9347044949506932.


AUC: 0.9291
F1: 0.6811 | Recall: 0.7477 | Precision: 0.6254

========== n_estimators 300 max_depth 8==========


[I 2026-09-11 15:11:39,722] Trial 2 finished with value: 0.926475505862709 and parameters: {'n_estimators': 300, 'max_depth': 8, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.85}. Best is trial 0 with value: 0.9347044949506932.


AUC: 0.9265
F1: 0.6724 | Recall: 0.7283 | Precision: 0.6245

========== n_estimators 300 max_depth 8==========


[I 2026-09-11 15:11:41,575] Trial 3 finished with value: 0.926475505862709 and parameters: {'n_estimators': 300, 'max_depth': 8, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.85}. Best is trial 0 with value: 0.9347044949506932.


AUC: 0.9265
F1: 0.6724 | Recall: 0.7283 | Precision: 0.6245

========== n_estimators 200 max_depth 8==========


[I 2026-09-11 15:11:42,916] Trial 4 finished with value: 0.9351018051879243 and parameters: {'n_estimators': 200, 'max_depth': 8, 'learning_rate': 0.05, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 4 with value: 0.9351018051879243.


AUC: 0.9351
F1: 0.6943 | Recall: 0.7898 | Precision: 0.6194

========== n_estimators 250 max_depth 8==========


[I 2026-09-11 15:11:44,640] Trial 5 finished with value: 0.9346321572328963 and parameters: {'n_estimators': 250, 'max_depth': 8, 'learning_rate': 0.05, 'subsample': 0.85, 'colsample_bytree': 0.8}. Best is trial 4 with value: 0.9351018051879243.


AUC: 0.9346
F1: 0.6899 | Recall: 0.7816 | Precision: 0.6175

========== n_estimators 100 max_depth 10==========


[I 2026-09-11 15:11:45,579] Trial 6 finished with value: 0.9312017113077585 and parameters: {'n_estimators': 100, 'max_depth': 10, 'learning_rate': 0.15, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 4 with value: 0.9351018051879243.


AUC: 0.9312
F1: 0.6836 | Recall: 0.7576 | Precision: 0.6227

========== n_estimators 150 max_depth 6==========


[I 2026-09-11 15:11:46,395] Trial 7 finished with value: 0.9351156796257445 and parameters: {'n_estimators': 150, 'max_depth': 6, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 7 with value: 0.9351156796257445.


AUC: 0.9351
F1: 0.6928 | Recall: 0.7863 | Precision: 0.6192

========== n_estimators 200 max_depth 6==========


[I 2026-09-11 15:11:47,266] Trial 8 finished with value: 0.9350126981114384 and parameters: {'n_estimators': 200, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.85, 'colsample_bytree': 0.8}. Best is trial 7 with value: 0.9351156796257445.


AUC: 0.9350
F1: 0.6913 | Recall: 0.7886 | Precision: 0.6153

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:11:47,569] Trial 9 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 300 max_depth 8==========


[I 2026-09-11 15:11:49,048] Trial 10 finished with value: 0.9300698325521284 and parameters: {'n_estimators': 300, 'max_depth': 8, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9301
F1: 0.6798 | Recall: 0.7465 | Precision: 0.6241

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:11:49,444] Trial 11 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:11:49,758] Trial 12 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 50 max_depth 4==========


[I 2026-09-11 15:11:49,991] Trial 13 finished with value: 0.9358281478945876 and parameters: {'n_estimators': 50, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9358
F1: 0.6880 | Recall: 0.8126 | Precision: 0.5965

========== n_estimators 150 max_depth 6==========


[I 2026-09-11 15:11:50,642] Trial 14 finished with value: 0.9360944806144801 and parameters: {'n_estimators': 150, 'max_depth': 6, 'learning_rate': 0.05, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9361
F1: 0.6901 | Recall: 0.8050 | Precision: 0.6039

========== n_estimators 50 max_depth 10==========


[I 2026-09-11 15:11:51,065] Trial 15 finished with value: 0.9344357615647148 and parameters: {'n_estimators': 50, 'max_depth': 10, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9344
F1: 0.6876 | Recall: 0.7769 | Precision: 0.6166

========== n_estimators 150 max_depth 4==========


[I 2026-09-11 15:11:51,582] Trial 16 finished with value: 0.9361120972670818 and parameters: {'n_estimators': 150, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9361
F1: 0.6867 | Recall: 0.8009 | Precision: 0.6011

========== n_estimators 100 max_depth 6==========


[I 2026-09-11 15:11:52,166] Trial 17 finished with value: 0.9358530018493603 and parameters: {'n_estimators': 100, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.85, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9359
F1: 0.6896 | Recall: 0.8044 | Precision: 0.6034

========== n_estimators 200 max_depth 6==========


[I 2026-09-11 15:11:52,981] Trial 18 finished with value: 0.9351483710492099 and parameters: {'n_estimators': 200, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9351
F1: 0.6943 | Recall: 0.7886 | Precision: 0.6202

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:11:53,271] Trial 19 finished with value: 0.9362080533026953 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.05, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9362
F1: 0.6850 | Recall: 0.7963 | Precision: 0.6010

========== n_estimators 100 max_depth 6==========


[I 2026-09-11 15:11:53,727] Trial 20 finished with value: 0.9359605728723605 and parameters: {'n_estimators': 100, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9360
F1: 0.6875 | Recall: 0.7998 | Precision: 0.6028

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:11:54,015] Trial 21 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 100 max_depth 4==========


[I 2026-09-11 15:11:54,524] Trial 22 finished with value: 0.9358344672950198 and parameters: {'n_estimators': 100, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9358
F1: 0.6865 | Recall: 0.8021 | Precision: 0.6001

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:11:54,807] Trial 23 finished with value: 0.936407026156528 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9364
F1: 0.6883 | Recall: 0.8062 | Precision: 0.6005

========== n_estimators 100 max_depth 8==========


[I 2026-09-11 15:11:55,494] Trial 24 finished with value: 0.9349753818753666 and parameters: {'n_estimators': 100, 'max_depth': 8, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9350
F1: 0.6927 | Recall: 0.7886 | Precision: 0.6176

========== n_estimators 100 max_depth 10==========


[I 2026-09-11 15:11:56,205] Trial 25 finished with value: 0.9326184573376218 and parameters: {'n_estimators': 100, 'max_depth': 10, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9326
F1: 0.6881 | Recall: 0.7711 | Precision: 0.6212

========== n_estimators 100 max_depth 6==========


[I 2026-09-11 15:11:56,668] Trial 26 finished with value: 0.9362277881900786 and parameters: {'n_estimators': 100, 'max_depth': 6, 'learning_rate': 0.05, 'subsample': 0.75, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9362
F1: 0.6868 | Recall: 0.8050 | Precision: 0.5989

========== n_estimators 100 max_depth 6==========


[I 2026-09-11 15:11:57,140] Trial 27 finished with value: 0.9359605728723605 and parameters: {'n_estimators': 100, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9360
F1: 0.6875 | Recall: 0.7998 | Precision: 0.6028

========== n_estimators 50 max_depth 8==========


[I 2026-09-11 15:11:57,594] Trial 28 finished with value: 0.9358423400676253 and parameters: {'n_estimators': 50, 'max_depth': 8, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9358
F1: 0.6934 | Recall: 0.7992 | Precision: 0.6124

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:11:57,946] Trial 29 finished with value: 0.9362989255748325 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.85, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9363
F1: 0.6858 | Recall: 0.8033 | Precision: 0.5983

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:11:58,278] Trial 30 finished with value: 0.9359094175023809 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.15, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9359
F1: 0.6912 | Recall: 0.8085 | Precision: 0.6036

========== n_estimators 50 max_depth 10==========


[I 2026-09-11 15:11:58,743] Trial 31 finished with value: 0.9349049505688731 and parameters: {'n_estimators': 50, 'max_depth': 10, 'learning_rate': 0.05, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9349
F1: 0.6880 | Recall: 0.7804 | Precision: 0.6151

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:11:59,047] Trial 32 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 50 max_depth 8==========


[I 2026-09-11 15:11:59,393] Trial 33 finished with value: 0.9357076556394189 and parameters: {'n_estimators': 50, 'max_depth': 8, 'learning_rate': 0.1, 'subsample': 0.85, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9357
F1: 0.6897 | Recall: 0.7933 | Precision: 0.6101

========== n_estimators 50 max_depth 4==========


[I 2026-09-11 15:11:59,624] Trial 34 finished with value: 0.9358281478945876 and parameters: {'n_estimators': 50, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9358
F1: 0.6880 | Recall: 0.8126 | Precision: 0.5965

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:11:59,978] Trial 35 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 100 max_depth 6==========


[I 2026-09-11 15:12:00,565] Trial 36 finished with value: 0.9352882451526315 and parameters: {'n_estimators': 100, 'max_depth': 6, 'learning_rate': 0.15, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9353
F1: 0.6890 | Recall: 0.7898 | Precision: 0.6110

========== n_estimators 150 max_depth 6==========


[I 2026-09-11 15:12:01,314] Trial 37 finished with value: 0.9356455207524874 and parameters: {'n_estimators': 150, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.85, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9356
F1: 0.6905 | Recall: 0.7927 | Precision: 0.6116

========== n_estimators 50 max_depth 8==========


[I 2026-09-11 15:12:01,754] Trial 38 finished with value: 0.9358423400676253 and parameters: {'n_estimators': 50, 'max_depth': 8, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9358
F1: 0.6934 | Recall: 0.7992 | Precision: 0.6124

========== n_estimators 150 max_depth 6==========


[I 2026-09-11 15:12:02,747] Trial 39 finished with value: 0.9356878148402965 and parameters: {'n_estimators': 150, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9357
F1: 0.6913 | Recall: 0.7945 | Precision: 0.6118

========== n_estimators 150 max_depth 4==========


[I 2026-09-11 15:12:03,316] Trial 40 finished with value: 0.936066414003622 and parameters: {'n_estimators': 150, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9361
F1: 0.6869 | Recall: 0.8033 | Precision: 0.5999

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:03,601] Trial 41 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:03,877] Trial 42 finished with value: 0.936407026156528 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9364
F1: 0.6883 | Recall: 0.8062 | Precision: 0.6005

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:04,162] Trial 43 finished with value: 0.9362080533026953 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.05, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9362
F1: 0.6850 | Recall: 0.7963 | Precision: 0.6010

========== n_estimators 100 max_depth 4==========


[I 2026-09-11 15:12:04,730] Trial 44 finished with value: 0.9359476869441046 and parameters: {'n_estimators': 100, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9359
F1: 0.6862 | Recall: 0.8103 | Precision: 0.5950

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:05,055] Trial 45 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 100 max_depth 6==========


[I 2026-09-11 15:12:05,506] Trial 46 finished with value: 0.9362567020948495 and parameters: {'n_estimators': 100, 'max_depth': 6, 'learning_rate': 0.05, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9363
F1: 0.6889 | Recall: 0.8050 | Precision: 0.6020

========== n_estimators 50 max_depth 10==========


[I 2026-09-11 15:12:05,950] Trial 47 finished with value: 0.9341755364217761 and parameters: {'n_estimators': 50, 'max_depth': 10, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9342
F1: 0.6880 | Recall: 0.7804 | Precision: 0.6151

========== n_estimators 150 max_depth 10==========


[I 2026-09-11 15:12:06,927] Trial 48 finished with value: 0.9281440394002965 and parameters: {'n_estimators': 150, 'max_depth': 10, 'learning_rate': 0.15, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9281
F1: 0.6741 | Recall: 0.7395 | Precision: 0.6194

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:07,231] Trial 49 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 100 max_depth 6==========


[I 2026-09-11 15:12:07,717] Trial 50 finished with value: 0.9358265239145882 and parameters: {'n_estimators': 100, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9358
F1: 0.6903 | Recall: 0.7992 | Precision: 0.6075

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:08,002] Trial 51 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:08,345] Trial 52 finished with value: 0.9359094175023809 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.15, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9359
F1: 0.6912 | Recall: 0.8085 | Precision: 0.6036

========== n_estimators 50 max_depth 10==========


[I 2026-09-11 15:12:08,831] Trial 53 finished with value: 0.9341755364217761 and parameters: {'n_estimators': 50, 'max_depth': 10, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9342
F1: 0.6880 | Recall: 0.7804 | Precision: 0.6151

========== n_estimators 100 max_depth 4==========


[I 2026-09-11 15:12:09,387] Trial 54 finished with value: 0.9359476869441046 and parameters: {'n_estimators': 100, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9359
F1: 0.6862 | Recall: 0.8103 | Precision: 0.5950

========== n_estimators 100 max_depth 10==========


[I 2026-09-11 15:12:10,192] Trial 55 finished with value: 0.9333144033751671 and parameters: {'n_estimators': 100, 'max_depth': 10, 'learning_rate': 0.1, 'subsample': 0.85, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9333
F1: 0.6865 | Recall: 0.7711 | Precision: 0.6186

========== n_estimators 50 max_depth 8==========


[I 2026-09-11 15:12:10,578] Trial 56 finished with value: 0.9359237508910709 and parameters: {'n_estimators': 50, 'max_depth': 8, 'learning_rate': 0.05, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9359
F1: 0.6895 | Recall: 0.7939 | Precision: 0.6094

========== n_estimators 150 max_depth 6==========


[I 2026-09-11 15:12:11,352] Trial 57 finished with value: 0.9356878148402965 and parameters: {'n_estimators': 150, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9357
F1: 0.6913 | Recall: 0.7945 | Precision: 0.6118

========== n_estimators 100 max_depth 6==========


[I 2026-09-11 15:12:11,979] Trial 58 finished with value: 0.9359605728723605 and parameters: {'n_estimators': 100, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9360
F1: 0.6875 | Recall: 0.7998 | Precision: 0.6028

========== n_estimators 150 max_depth 6==========


[I 2026-09-11 15:12:12,745] Trial 59 finished with value: 0.9356878148402965 and parameters: {'n_estimators': 150, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9357
F1: 0.6913 | Recall: 0.7945 | Precision: 0.6118

========== n_estimators 50 max_depth 10==========


[I 2026-09-11 15:12:13,221] Trial 60 finished with value: 0.9332061262739064 and parameters: {'n_estimators': 50, 'max_depth': 10, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9332
F1: 0.6877 | Recall: 0.7728 | Precision: 0.6194

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:13,536] Trial 61 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:13,868] Trial 62 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 300 max_depth 6==========


[I 2026-09-11 15:12:15,340] Trial 63 finished with value: 0.9358173095932876 and parameters: {'n_estimators': 300, 'max_depth': 6, 'learning_rate': 0.05, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9358
F1: 0.6958 | Recall: 0.8015 | Precision: 0.6147

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:15,712] Trial 64 finished with value: 0.936407026156528 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9364
F1: 0.6883 | Recall: 0.8062 | Precision: 0.6005

========== n_estimators 300 max_depth 10==========


[I 2026-09-11 15:12:18,110] Trial 65 finished with value: 0.9220226939201578 and parameters: {'n_estimators': 300, 'max_depth': 10, 'learning_rate': 0.15, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9220
F1: 0.6596 | Recall: 0.7102 | Precision: 0.6157

========== n_estimators 50 max_depth 10==========


[I 2026-09-11 15:12:18,778] Trial 66 finished with value: 0.9343213062786738 and parameters: {'n_estimators': 50, 'max_depth': 10, 'learning_rate': 0.1, 'subsample': 0.85, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9343
F1: 0.6906 | Recall: 0.7840 | Precision: 0.6171

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:19,184] Trial 67 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:19,511] Trial 68 finished with value: 0.9358926128397791 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.15, 'subsample': 0.85, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9359
F1: 0.6868 | Recall: 0.7992 | Precision: 0.6021

========== n_estimators 100 max_depth 10==========


[I 2026-09-11 15:12:20,447] Trial 69 finished with value: 0.9349406075210327 and parameters: {'n_estimators': 100, 'max_depth': 10, 'learning_rate': 0.05, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9349
F1: 0.6887 | Recall: 0.7799 | Precision: 0.6167

========== n_estimators 100 max_depth 10==========


[I 2026-09-11 15:12:21,386] Trial 70 finished with value: 0.9347782095210981 and parameters: {'n_estimators': 100, 'max_depth': 10, 'learning_rate': 0.05, 'subsample': 0.85, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9348
F1: 0.6901 | Recall: 0.7828 | Precision: 0.6170

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:21,737] Trial 71 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 50 max_depth 8==========


[I 2026-09-11 15:12:22,144] Trial 72 finished with value: 0.9352321472348282 and parameters: {'n_estimators': 50, 'max_depth': 8, 'learning_rate': 0.15, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9352
F1: 0.6910 | Recall: 0.7968 | Precision: 0.6100

========== n_estimators 100 max_depth 6==========


[I 2026-09-11 15:12:22,729] Trial 73 finished with value: 0.9359605728723605 and parameters: {'n_estimators': 100, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9360
F1: 0.6875 | Recall: 0.7998 | Precision: 0.6028

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:23,089] Trial 74 finished with value: 0.9359094175023809 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.15, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9359
F1: 0.6912 | Recall: 0.8085 | Precision: 0.6036

========== n_estimators 300 max_depth 4==========


[I 2026-09-11 15:12:24,344] Trial 75 finished with value: 0.9362119720370414 and parameters: {'n_estimators': 300, 'max_depth': 4, 'learning_rate': 0.05, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9362
F1: 0.6872 | Recall: 0.8033 | Precision: 0.6004

========== n_estimators 200 max_depth 4==========


[I 2026-09-11 15:12:25,196] Trial 76 finished with value: 0.9361579923540198 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9362
F1: 0.6870 | Recall: 0.7986 | Precision: 0.6027

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:25,532] Trial 77 finished with value: 0.9362413095887687 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.05, 'subsample': 0.85, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9362
F1: 0.6856 | Recall: 0.7980 | Precision: 0.6010

========== n_estimators 50 max_depth 4==========


[I 2026-09-11 15:12:25,861] Trial 78 finished with value: 0.9358281478945876 and parameters: {'n_estimators': 50, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9358
F1: 0.6880 | Recall: 0.8126 | Precision: 0.5965

========== n_estimators 50 max_depth 4==========


[I 2026-09-11 15:12:26,173] Trial 79 finished with value: 0.9358145205841584 and parameters: {'n_estimators': 50, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.85, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9358
F1: 0.6870 | Recall: 0.8121 | Precision: 0.5953

========== n_estimators 100 max_depth 6==========


[I 2026-09-11 15:12:26,705] Trial 80 finished with value: 0.9359605728723605 and parameters: {'n_estimators': 100, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9360
F1: 0.6875 | Recall: 0.7998 | Precision: 0.6028

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:26,985] Trial 81 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:27,357] Trial 82 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 100 max_depth 8==========


[I 2026-09-11 15:12:27,953] Trial 83 finished with value: 0.9361872239940079 and parameters: {'n_estimators': 100, 'max_depth': 8, 'learning_rate': 0.05, 'subsample': 0.8, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9362
F1: 0.6920 | Recall: 0.7992 | Precision: 0.6102

========== n_estimators 50 max_depth 8==========


[I 2026-09-11 15:12:28,409] Trial 84 finished with value: 0.9358423400676253 and parameters: {'n_estimators': 50, 'max_depth': 8, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9358
F1: 0.6934 | Recall: 0.7992 | Precision: 0.6124

========== n_estimators 50 max_depth 4==========


[I 2026-09-11 15:12:28,700] Trial 85 finished with value: 0.9359014741219494 and parameters: {'n_estimators': 50, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9359
F1: 0.6863 | Recall: 0.8126 | Precision: 0.5939

========== n_estimators 100 max_depth 8==========


[I 2026-09-11 15:12:29,317] Trial 86 finished with value: 0.9349753818753666 and parameters: {'n_estimators': 100, 'max_depth': 8, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9350
F1: 0.6927 | Recall: 0.7886 | Precision: 0.6176

========== n_estimators 50 max_depth 4==========


[I 2026-09-11 15:12:29,567] Trial 87 finished with value: 0.9350136866210031 and parameters: {'n_estimators': 50, 'max_depth': 4, 'learning_rate': 0.05, 'subsample': 0.8, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9350
F1: 0.6880 | Recall: 0.8238 | Precision: 0.5907

========== n_estimators 250 max_depth 4==========


[I 2026-09-11 15:12:30,531] Trial 88 finished with value: 0.9361728199974919 and parameters: {'n_estimators': 250, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.85, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9362
F1: 0.6876 | Recall: 0.7939 | Precision: 0.6064

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:30,949] Trial 89 finished with value: 0.9362989255748325 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.85, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9363
F1: 0.6858 | Recall: 0.8033 | Precision: 0.5983

========== n_estimators 100 max_depth 6==========


[I 2026-09-11 15:12:31,415] Trial 90 finished with value: 0.9358530018493603 and parameters: {'n_estimators': 100, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.85, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9359
F1: 0.6896 | Recall: 0.8044 | Precision: 0.6034

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:31,692] Trial 91 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 50 max_depth 6==========


[I 2026-09-11 15:12:31,998] Trial 92 finished with value: 0.936588311749933 and parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9366
F1: 0.6890 | Recall: 0.8068 | Precision: 0.6012

========== n_estimators 100 max_depth 4==========


[I 2026-09-11 15:12:32,420] Trial 93 finished with value: 0.9356845315763848 and parameters: {'n_estimators': 100, 'max_depth': 4, 'learning_rate': 0.05, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9357
F1: 0.6860 | Recall: 0.8121 | Precision: 0.5938

========== n_estimators 100 max_depth 6==========


[I 2026-09-11 15:12:32,951] Trial 94 finished with value: 0.9361977445600906 and parameters: {'n_estimators': 100, 'max_depth': 6, 'learning_rate': 0.05, 'subsample': 0.85, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9362
F1: 0.6894 | Recall: 0.8062 | Precision: 0.6021

========== n_estimators 50 max_depth 8==========


[I 2026-09-11 15:12:33,317] Trial 95 finished with value: 0.9356973822007274 and parameters: {'n_estimators': 50, 'max_depth': 8, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9357
F1: 0.6924 | Recall: 0.7992 | Precision: 0.6107

========== n_estimators 200 max_depth 4==========


[I 2026-09-11 15:12:34,048] Trial 96 finished with value: 0.9359955237462594 and parameters: {'n_estimators': 200, 'max_depth': 4, 'learning_rate': 0.15, 'subsample': 0.85, 'colsample_bytree': 0.75}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9360
F1: 0.6899 | Recall: 0.7945 | Precision: 0.6096

========== n_estimators 100 max_depth 4==========


[I 2026-09-11 15:12:34,463] Trial 97 finished with value: 0.9361220176666429 and parameters: {'n_estimators': 100, 'max_depth': 4, 'learning_rate': 0.1, 'subsample': 0.75, 'colsample_bytree': 0.85}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9361
F1: 0.6863 | Recall: 0.8109 | Precision: 0.5949

========== n_estimators 50 max_depth 4==========


[I 2026-09-11 15:12:34,700] Trial 98 finished with value: 0.9348667517349755 and parameters: {'n_estimators': 50, 'max_depth': 4, 'learning_rate': 0.05, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9349
F1: 0.6883 | Recall: 0.8244 | Precision: 0.5909

========== n_estimators 300 max_depth 10==========


[I 2026-09-11 15:12:36,551] Trial 99 finished with value: 0.9216520028333509 and parameters: {'n_estimators': 300, 'max_depth': 10, 'learning_rate': 0.15, 'subsample': 0.75, 'colsample_bytree': 0.8}. Best is trial 9 with value: 0.936588311749933.


AUC: 0.9217
F1: 0.6587 | Recall: 0.7096 | Precision: 0.6146


In [8]:
print(f"Best Score: {study.best_value}")
print(f"Best Parameters: {study.best_params}")
best = study.best_trial

print(f"Trial Number: {best.number}")
print(f"Best Score: {best.value}")
print(f"Best Params: {best.params}")


Best Score: 0.936588311749933
Best Parameters: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}
Trial Number: 9
Best Score: 0.936588311749933
Best Params: {'n_estimators': 50, 'max_depth': 6, 'learning_rate': 0.1, 'subsample': 0.8, 'colsample_bytree': 0.75}


### Without oversampling (10000 samples)
- Avg AUC: 0.7961
- Avg F1: 0.6768
- Avg Recall: 0.6476
- Avg Precision: 0.7091
- Std AUC: 0.011227993955145489

### With SMOTE 
- (50000 samples)
- Avg AUC: 0.9380
- Avg F1: 0.6928
- Avg Recall: 0.6803
- Avg Precision: 0.7061
- Std AUC: 0.0027307035663074416

- (100000 samples)
- Avg AUC: 0.9366
- Avg F1: 0.7012
- Avg Recall: 0.7416
- Avg Precision: 0.6650
- Std AUC: 0.001856941113581715

(200000 samples)
- Avg AUC: 0.9399
- Avg F1: 0.6997
- Avg Recall: 0.6903
- Avg Precision: 0.7095
- Std AUC: 0.0008880157870868589

(500000 samples)
- Avg AUC: 0.9402
- Avg F1: 0.6996
- Avg Recall: 0.6896
- Avg Precision: 0.7098
- Std AUC: 0.0008942851500033605

### With ADASYN 
- (50000 samples)
- Avg AUC: 0.9379
- Avg F1: 0.6930
- Avg Recall: 0.6819
- Avg Precision: 0.7048
- Std AUC: 0.0026846216264028214